src/ConnectionRequest.jl

```diff
@@ -49,7 +50,7 @@ function request(::Type{ConnectionPoolLayer{Next}}, url::URI, req, body;
         close(io)
         rethrow(isioerror(e) ? IOError(e, "during request($url)") : e)
     finally
-        if proxy !== nothing && target_url.scheme == "https"
+        if (proxy !== nothing && target_url.scheme == "https") || reuse_limit == 0
```

Comment:
@samoconnor, in this case, if the `reuse_limit=0`, then the connection was still hanging around in the pool until the next request, at which point it realized it couldn't be re-used and closed it. This just ensures that if a connection shouldn't be re-used, it is closed immediately after.

Reply:
Could you please try to maintain 80-col wrapping in files that currently adhere to that.